### ISSUE TITLE
Dragonflight Launch - Remove Code for Wholly Unsupported Specs

### ISSUE THREAD

**Issue by emallson** [WoWAnalyzer/WoWAnalyzer#5090]
[Documentation](https://github.com/WoWAnalyzer/WoWAnalyzer/wiki/Enabling-a-Spec-for-Dragonflight)

- [ ] Remove the `CombatLogParser` for each retail spec that hasn't received a compatibility update for Dragonflight (full support not required)
- [ ] Remove the `tsconfig.json` exclusions for all specs/classes.
- [ ] Check that this correctly:
  - [ ] Shows the spec on the Specs page as unsupported
  - [ ] Does not allow clicking through to an example log
  - [ ] Does not crash when visiting an unsupported spec from report character selection

**Comment by ToppleTheNun** [comment]
I worked on this a bit and have a few notes:
- `parser` is currently required in `Config`. You can work around this by marking `parser` as optional in the interface. You then need to update `useParser` to conditionally invoke the function.
- Once the `tsconfig.json` exclusions are removed and you re-add the specs into `src/parser/index.ts`, you get a _bunch_ of errors from the freshly enabled specs. See below for an example: 
![image](https://user-images.githubusercontent.com/1672786/199144890-b54d82bb-d28e-435f-937e-130dcf412f00.png)
- Specs won't show up on the specializations page unless their expansion is set to `Expansion.Dragonflight` in their `Config`, which I took a swing at updating a few in #5279.

**Comment by emallson** [comment]
I did a pass on this before my trip and ran into similar issues. I ended up commenting out the changelogs and combatlogparsers in particular, just leaving the config as part of the import tree. That way we can list the spec but not have to update the contents yet. Before doing that, though, I need to figure out what the updated documentation looks like.

It will probably also wait until #5091 gets some work done so we can better show totally unsupported specs.
